### DISCUSSION TITLE
Deleting 'pending' transactions / transfers reloading

### DISCUSSION THREAD

**Discussion by alexchatwin** [firefly-iii/firefly-iii#8718] [Firefly III] [answered]
**Situation**:
Hi, I'm using the gocardless automation to pull from my accounts. 

I had an issue where it was pulling duplicates with booked vs pending status (I basically never want to see pending).  https://github.com/orgs/firefly-iii/discussions/8276#discussioncomment-8346096

I setup an automation to delete these, then I also automatically call the purge api to remove the deletions, so that the booked transactions don't get scrubbed as duplicates by the importer

I _also_, separately delete one side of any transfer transactions - these have to be manually matched as the bank doesn't provide consistent data on both sides.

**Problem**:
When I reload to get booked transactions, I also get reloads of the deleted transfer (cos I'm purging deletes)

I know I'm deep into 'messing around' territory, can anyone think of a better way to achieve my desired state - no pending vs booked conflicts, deleted transfers not 'rising from the dead'?

**Comment by JC5** [comment]
I'm actually not aware of transactions that are "pending" being duplicate with the ones that are "booked", is that something I should look into?

**Reply by alexchatwin** [reply]
Possibly. I can disable my deletion and try and grab some examples for you

From distant memory, I think it was something to do with weekends and the booked-ness applying overnight.

**Reply by JC5** [reply]
Let me know, I'm curious to here more about it.

**Reply by alexchatwin** [reply]
Here you go @JC5 

![image](https://github.com/firefly-iii/firefly-iii/assets/22892404/0533719d-5414-486f-938f-ef710809c397)
![image](https://github.com/firefly-iii/firefly-iii/assets/22892404/2c9b0d65-6329-4aa4-bb92-f9366f0e780e)

There's not much way you'd know these are the same txn, except spending 70p at amazon is so unusual it stuck out on the log

**Reply by alexchatwin** [reply]
I bascially want to bin all 'pending' - they just mess up the data

**Reply by digitlength** [reply]
Your pending and booked transactions look identical in form to ones produced by my bank, imported via GoCardless.

The pending transaction has no description but a destination account.

The booked transaction has a description which is not identical to anything in the pending transaction. Ditto the destination account.

I just delete the pending transaction once the booked one has come through. I don't do any extra purging. When the importer runs again, it correctly does not re-import the (now deleted) pending transaction, nor the still existing booked transaction.

**Reply by alexchatwin** [reply]
Thanks, so if I just delete pending, I should be ok?

**Reply by JC5** [reply]
Yes, you should be fine. You can even make a rule do it for you.

**Reply by digitlength** [reply]
Yep, should be. I guess it'll also depend on your import settings, especially around duplicate detection. If you've left this as default then you should be fine.

**Reply by digitlength** [reply]
> You can even make a rule do it for you.

If you mean a rule to delete transaction because it is pending then agreed. If you mean a rule to delete a pending transaction once it's booked twin comes in then I'm all ears - I've currently got a webhook doing some dirty stuff to make this work, but if it can be done in rules then I'd much prefer that!

**Reply by JC5** [reply]
If a booked transaction is *not ever* a duplicate of a pending transaction (because the transaction identifier changes!), then it's enough to make a rule to delete all transactions with the "pending" tag.

**Reply by alexchatwin** [reply]
This is my next investigation.. I have a number of pendings now.. waiting to see what happens to them when the book

**Reply by digitlength** [reply]
>...then it's enough to make a rule to delete all transactions with the "pending" tag

Agreed, and understood.

I'm interested in up-to-date status, and sometimes the booked-twin does not come in until a day or so after the pending transaction, and in the case of some retailers it could be some days later. In this case I want the pending transaction to remain as it ensures my asset account balance is correct (because the value of the pending transaction is not available for me to spend in real life as my bank has 'reserved' it). As a result I only want to delete the pending transaction when the booked-twin appears.

I'll continue to use my webhook approach!

**Reply by alexchatwin** [reply]
Ok, looks like the pending and booked remain unrelated! I'll keep checking, but it looks like I can just delete them

**Comment by kesot** [comment]
If you use gocardless for import switch to external id duplicate detection, these transactions you are describing should have the same id so they will not be duplicated when then change the state

**Reply by JC5** [reply]
Mmm, that differs per bank, I've heard people say that these IDs have changed after a transaction went from "pending" to "booked".

But yes, duplicate detection will stop this from working.

**Comment by karl0ss** [comment]
Did we ever work this out? as I have similar situation with my gocardless/santander account 


![image](https://github.com/user-attachments/assets/975062d8-dc97-42ad-a11e-bd1a74962e6b)
This one comes in first "pending" in my account

![image](https://github.com/user-attachments/assets/a2c9178e-4b39-4372-9483-f9bfc711add1)
then updated on the next run (day) same transaction but has now updated it details, but now is duplicated...

any idea on this?

thanks

**Reply by JC5** [reply, marked as answer]
Delete the pending transactions with a rule.

**Reply by alexchatwin** [reply]
Yes, this worked

**Reply by karl0ss** [reply]
Not sure this work work for me, as if you notice, the second screenshot, although is correct and has the correct description, and is no longer pending in my account, seems to still be tagged with pending? so if i delete pending, then i would loose both transactions and the complete record?

maybe something else is wrong on why it still has it marked as pending on the 7am import?

**Reply by JC5** [reply]
You would not lose the other transaction
